testingutils/setup.go

```diff
@@ -39,33 +39,49 @@ func RunSmartContractMigrations() {
 	if isRunningOnCI {
 		return
 	}
+
+	var err error
 	projDir := GetProjectDir()
 	migrationScript := path.Join(projDir, "build", "scripts", "migrate.sh")
-	_, err := exec.Command(migrationScript, projDir).Output()
-	if err != nil {
-		log.Fatal(err)
+	for i := 0; i < 3; i++ {
+		log.Infof("Trying to migrate contracts for the %d th time", i)
+		_, err = exec.Command(migrationScript, projDir).Output()
+		if err == nil {
+			return
+		}
 	}
+	// trying 3 times to migrate didnt work
+	log.Fatal(err)
 }
 
 // GetSmartContractAddresses finds migrated smart contract addresses for localgeth
 func GetSmartContractAddresses() *config.SmartContractAddresses {
-	dat, err := findContractDeployJSON()
+	iddat, err := findContractDeployJSON("IdentityFactory.json")
+	if err != nil {
+		panic(err)
+	}
+
+	ancdat, err := findContractDeployJSON("AnchorRepository.json")
+	if err != nil {
+		panic(err)
+	}
+
+	payobdat, err := findContractDeployJSON("PaymentObligation.json")
 	if err != nil {
 		panic(err)
 	}
-	idFactoryAddrOp := getOpForContract(".contracts.IdentityFactory.address")
-	anchorRepoAddrOp := getOpForContract(".contracts.AnchorRepository.address")
-	payObAddrOp := getOpForContract(".contracts.PaymentObligation.address")
+
+	addrOp := getOpForContract(".networks.8383.address")
```

Comment:
with this hardcoded network ID we will not be able to run integration/testworld tests against rinkeby/ropsten...

Reply:
We don't migrate contracts for other networks, its read from config, so it should be fine.